src/vl-map-draw-action.js

```diff
@@ -22,4 +24,38 @@ export class VlMapDrawAction extends VlMapLayerAction {
   onDraw(callback) {
     this.__callback = callback;
   }
+
+  get __drawOptions() {
+    if (this.dataset.vlSnapping !== undefined) {
+      const snappingLayers = this.__snappingLayers;
+      if (snappingLayers.length == 0) {
+        return {snapping: true};
+      } else {
+        return {
+          snapping: {
+            layer: this.__createSnappingLayer(),
+            pixelTolerance: this.dataset.vlSnappingPixelTolerance || 10,
+            node: false,
+            vertex: false,
+          },
+        };
+      }
+    } else {
+      return {};
+    }
+  }
+
+  __createSnappingLayer() {
+    const snappingLayer = new VlCompositeVectorLayer(this.__snappingLayers.map((layer) => layer._layer), {});
+    const firstVectorLayer = this.__snappingLayers[0];
+    snappingLayer.setStyle(firstVectorLayer.style);
+    firstVectorLayer.addEventListener(VlMapVectorLayer.EVENTS.styleChanged, (event) => {
+      snappingLayer.setStyle(event.target.style);
+    });
```

Comment:
Goed genoeg in kader van de Voortoets, vermoedelijk kunnen we dit later uitbreiden met een attribuut zodat we declaraties kunnen aanduiden van welke kaartlaag de stijl wordt genomen, daar het nu een beetje automatisch de eerste kaartlaag is.